src/test/unit/algorithms/SpatialPoolerTest.cpp

```diff
@@ -126,118 +126,6 @@ namespace {
     return true;
   }
 
-  void check_spatial_eq(SpatialPooler sp1, SpatialPooler sp2)
-  {
-    UInt numColumns = sp1.getNumColumns();
-    UInt numInputs = sp2.getNumInputs();
-
-    ASSERT_TRUE(sp1.getNumColumns() == sp2.getNumColumns());
-    ASSERT_TRUE(sp1.getNumInputs() == sp2.getNumInputs());
-    ASSERT_TRUE(sp1.getPotentialRadius() ==
-              sp2.getPotentialRadius());
-    ASSERT_TRUE(sp1.getPotentialPct() == sp2.getPotentialPct());
-    ASSERT_TRUE(sp1.getGlobalInhibition() ==
-              sp2.getGlobalInhibition());
-    ASSERT_TRUE(sp1.getNumActiveColumnsPerInhArea() ==
-              sp2.getNumActiveColumnsPerInhArea());
-    ASSERT_TRUE(almost_eq(sp1.getLocalAreaDensity(),
-              sp2.getLocalAreaDensity()));
-    ASSERT_TRUE(sp1.getStimulusThreshold() ==
-              sp2.getStimulusThreshold());
-    ASSERT_TRUE(sp1.getDutyCyclePeriod() == sp2.getDutyCyclePeriod());
-    ASSERT_TRUE(almost_eq(sp1.getBoostStrength(), sp2.getBoostStrength()));
-    ASSERT_TRUE(sp1.getIterationNum() == sp2.getIterationNum());
-    ASSERT_TRUE(sp1.getIterationLearnNum() ==
-              sp2.getIterationLearnNum());
-    ASSERT_TRUE(sp1.getSpVerbosity() == sp2.getSpVerbosity());
-    ASSERT_TRUE(sp1.getWrapAround() == sp2.getWrapAround());
-    ASSERT_TRUE(sp1.getUpdatePeriod() == sp2.getUpdatePeriod());
-    ASSERT_TRUE(almost_eq(sp1.getSynPermTrimThreshold(),
-              sp2.getSynPermTrimThreshold()));
-    cout << "check: " << sp1.getSynPermActiveInc() << " " <<
-      sp2.getSynPermActiveInc() << endl;
-    ASSERT_TRUE(almost_eq(sp1.getSynPermActiveInc(),
-              sp2.getSynPermActiveInc()));
-    ASSERT_TRUE(almost_eq(sp1.getSynPermInactiveDec(),
-              sp2.getSynPermInactiveDec()));
-    ASSERT_TRUE(almost_eq(sp1.getSynPermBelowStimulusInc(),
-              sp2.getSynPermBelowStimulusInc()));
-    ASSERT_TRUE(almost_eq(sp1.getSynPermConnected(),
-              sp2.getSynPermConnected()));
-    ASSERT_TRUE(almost_eq(sp1.getMinPctOverlapDutyCycles(),
-              sp2.getMinPctOverlapDutyCycles()));
-
-
-    auto boostFactors1 = new Real[numColumns];
-    auto boostFactors2 = new Real[numColumns];
-    sp1.getBoostFactors(boostFactors1);
-    sp2.getBoostFactors(boostFactors2);
-    ASSERT_TRUE(check_vector_eq(boostFactors1, boostFactors2, numColumns));
```

Comment:
Another function is `check_vector_eq();` used in two scenarios:
- comparing `array[]` with `vector<>`. this could be avoided ei by `vector.data()`
- for 2 vectors of Real, it calls `almost_eq` element-wise; I'm not sure if that can be done implicitly without the need for such function?

`check_vector_eq` should be implemented as `std::equal()` with the "twice" example:
http://www.tenouk.com/cpluscodesnippet/cplusstlvectoralgorithmequal.html